src/operations/skills_reader.rs

```diff
@@ -0,0 +1,590 @@
+
+    // If the skills directory doesn't exist, return empty list
+    if !skills_dir.exists() || !skills_dir.is_dir() {
+        return Ok(Vec::new());
+    }
+
+    let mut skill_folders = Vec::new();
+
+    for entry in fs::read_dir(&skills_dir)
+        .with_context(|| format!("Failed to read skills directory: {}", skills_dir.display()))?
+    {
+        let entry = entry?;
+        let path = entry.path();
+
+        // Skip non-directories
+        if !path.is_dir() {
+            continue;
+        }
+
+        // Check if SKILL.md exists in this folder
+        let skill_file = path.join(SKILL_FILENAME);
+        if !skill_file.exists() {
+            continue;
+        }
+
+        // Get the folder name
+        if let Some(folder_name) = path.file_name().and_then(|n| n.to_str()) {
+            let relative_path = PathBuf::from(AI_RULE_SOURCE_DIR)
+                .join(SKILLS_DIR)
+                .join(folder_name);
+
+            skill_folders.push(SkillFolder {
+                name: folder_name.to_string(),
+                relative_path,
+                full_path: path,
+            });
+        }
+    }
+
+    Ok(skill_folders)
+}
+
+/// Creates symlinks for each skill folder in the target directory
+#[allow(dead_code)]
+pub fn create_skill_symlinks(current_dir: &Path, target_dir: &str) -> Result<Vec<PathBuf>> {
+    let skill_folders = find_skill_folders(current_dir)?;
+
+    if skill_folders.is_empty() {
+        return Ok(Vec::new());
+    }
+
+    let mut created_symlinks = Vec::new();
+
+    for skill in skill_folders {
+        // Create symlink name with prefix: ai-rules-generated-<name>
+        let symlink_name = format!("{}{}", GENERATED_FILE_PREFIX, skill.name);
+        let from_path = PathBuf::from(target_dir).join(&symlink_name);
+
+        // Calculate the relative path from symlink location to source
+        let relative_source = calculate_relative_path(&from_path, &skill.relative_path);
+
+        // Create the actual symlink
+        let symlink_path = current_dir.join(&from_path);
+        create_relative_symlink(&symlink_path, &relative_source)?;
+
+        created_symlinks.push(symlink_path);
+    }
+
+    Ok(created_symlinks)
+}
+
+/// Removes generated skill symlinks from target directory
+#[allow(dead_code)]
+pub fn remove_generated_skill_symlinks(current_dir: &Path, target_dir: &str) -> Result<()> {
+    let target_path = current_dir.join(target_dir);
+
+    if !target_path.exists() {
+        return Ok(());
+    }
+
+    for entry in fs::read_dir(&target_path)? {
+        let entry = entry?;
+        let path = entry.path();
+
+        if let Some(file_name) = path.file_name().and_then(|n| n.to_str()) {
+            // Only remove entries that start with our generated prefix
+            // AND are symlinks (to avoid accidentally removing real folders)
+            if file_name.starts_with(GENERATED_FILE_PREFIX) && path.is_symlink() {
```

Comment:
Think we can remove the symlink check, sometimes things get lost/moved etc and better to do cleanup